### DISCUSSION TITLE
Skip running workflows

### DISCUSSION THREAD

**Discussion by dentuzhik** [community/community#27081] [Actions]
<p>Hello!</p>
<p>We would like to build a fairly unsophisticated set of actions to automate release pipelines, based on pull request coments, and currently we are utilising the syntax to be able to have conditional jobs.</p>
<pre><code>name: Complete Release

on:
  issue_comment:

jobs:
  merge-release-into-master:
    runs-on: ubuntu-latest
    name: Merge release branch into master
    if: github.event.comment &amp;&amp; github.event.comment.body == 'release-complete'
</code></pre>
<p>It’s working fairly well, with exception on how Github handles such runs. Used syntax means that this workflow is triggered on a comment <strong>for every issue and PR</strong>. </p>
<p>Before we saw these workflows as “skipped”. You couldn’t really access the workflow run, clicking on it would show you that workflow never started. Since some recent changes, these skipped runs are starting to show as failed and we started to get notifications from those.</p>
<p>It would be great though, if there would be a way to completely prevent workflow from running, or at least exclude such “shadow runs” from UI. </p>
<p>Syntax could look something like this:</p>
<pre><code>name: Complete Release

on:
  issue_comment:
    if: github.event.comment &amp;&amp; github.event.comment.body == 'release-complete'

jobs:
  merge-release-into-master:
    runs-on: ubuntu-latest
    name: Merge release branch into master
</code></pre>
<p>Every skipped workflow takes some amount of time to run, for us it’s around 3s. It’s hard to say what’s happening, but in case if it’s spinning off VM and consuming concurrency jobs limit, it would quickly become a problem for a large repo with lots of comments and other workflows. Statement above is rather a speculation, so it be great to get some clarification on this to understand what’s happening under the hood for such workflows.</p>
<p>Thank you!</p>
<p><a href="https://github.community/t/conditional-jobs-based-on-on-conditions/16729">Related Thread</a></p>

**Comment by Yanjingzhu** [comment]
<p>Sorry for the current behavior of skipped workflow. We have reported the feedback to the appropriate engineering team for further investigation and evaluation. If have any progress, we will notify you in time. </p>
<p>In addition, there is another ticket which reported the same issue: <br>
<a href="https://github.community/t/status-of-workflows-with-no-running-jobs/16782">https://github.community/t5/GitHub-Actions/Status-of-workflows-with-no-running-jobs/td-p/37160</a></p>

**Comment by jsoref** [comment, marked as answer]
<p>Fwiw, this should be fixed per:</p><aside class="quote" data-post="12" data-topic="16782">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/status-of-workflows-with-no-running-jobs/16782/12">Status of workflows with no running jobs</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    Thanks for your feedback! 
A workflow with no jobs should now correctly be marked as “skipped” instead of failed.
  </blockquote>
</aside>
